internal/contour/translator_test.go

```diff
@@ -806,7 +809,76 @@ func TestTranslatorAddIngress(t *testing.T) {
 			Domains: []string{"*"},
 			Routes: []*v2.Route{{
 				Match:  prefixmatch("/hello"),
-				Action: clusteraction("default/hello/80"),
+				Action: clusteraction("default/hello/80", nil),
+			}},
+		}},
+		ingress_https: []*v2.VirtualHost{},
+	}, {
+		name: "explicitly set upstream timeout to seconds",
+		ing: &v1beta1.Ingress{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "correct",
+				Namespace: "default",
+				Annotations: map[string]string{
+					"contour.heptio.com/request-timeout": "20s",
+				},
+			},
+			Spec: v1beta1.IngressSpec{
+				Backend: backend("backend", intstr.FromInt(80)),
+			},
+		},
+		ingress_http: []*v2.VirtualHost{{
+			Name:    "*",
+			Domains: []string{"*"},
+			Routes: []*v2.Route{{
+				Match:  prefixmatch("/"), // match all
+				Action: clusteraction("default/backend/80", &duration20seconds),
```

Comment:
use the helper I suggested above it should do something like

func clusteractiontimeout(cluster string, timeout time.Duration) ... {
    c := clusteraction(cluster)
    c.Timeout = &timeout // or whatever field we need
    return c
}